File: marfim-web/src/model/Organization.ts
export default class Organization {
  public id?: number;

  public cnpj: string;

  public name: string;

  public description?: string;

  public avatarUrl?: string;

  public createdAt?: Date;

  public updatedAt?: Date;

  public enabled?: boolean;

  constructor() {
    this.name = '';
    this.cnpj = '';
    this.description = '';
    this.avatarUrl = '';
  }
}

export interface ICreateUpdateOrganization {
  cnpj: string;
  name: string;
  description?: string;
  avatarUrl?: string;
}

export interface IOrganization {
  id?: number;
  name: string;
  avatarUrl?: string;
}

File: marfim-api/src/main/java/org/ipdec/marfim/api/model/Organization.java
package org.ipdec.marfim.api.model;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.*;

import javax.persistence.Entity;
import javax.persistence.Table;
import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDateTime;


@Data
@Entity
@DynamicInsert @DynamicUpdate
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "organization", schema = "marfim")
@Where(clause = "enabled=true")
public class Organization implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String cnpj;

    @Column(nullable = false, unique = true)
    private String name;

    @Column
    private String description;

    @Column
    private String avatarUrl;

    @Column
    @CreationTimestamp
    private LocalDateTime createdAt;

    @Column
    @UpdateTimestamp
    private LocalDateTime updatedAt;

    @Column
    private Boolean enabled = true;

}
